sklearn/semi_supervised/self_training.py

```diff
@@ -0,0 +1,156 @@
+import numpy as np
+
+from ..base import BaseEstimator, ClassifierMixin
+from ..utils.validation import check_X_y, check_array, check_is_fitted
+from ..utils.metaestimators import if_delegate_has_method
+from ..utils import safe_mask
+
+
+def _check_estimator(estimator):
+    """Make sure that an estimator implements the necessary methods."""
+    if not hasattr(estimator, "predict_proba"):
+        raise ValueError("The base_estimator should implement predict_proba!")
+
+
+class SelfTraining(BaseEstimator, ClassifierMixin):
+
+    """Self-training classifier
+
+    Parameters
+    ----------
+    base_estimator : estimator object
+        An estimator object implementing `fit` and `predict_proba`.
+
+    threshold : float
+        Threshold above which predictions are added to the labeled dataset.
+        Should be in [0, 1].
```

Comment:
What's the point of including 1? Should this be `[0, 1)`?

Reply:
Yes I agree